File: serve/router/yipan.js
const express = require('express')
const router = express.Router()

const yipan_handler = require('../router_handler/yipan')

router.post('/add', yipan_handler.add)
router.post('/detail', yipan_handler.detail)
router.post('/update', yipan_handler.update)
router.post('/list', yipan_handler.list)
router.post('/start', yipan_handler.startChairside)
router.post('/complete', yipan_handler.completeChairside)
router.post('/history', yipan_handler.getHistory)
router.post('/updateChairsideVideo', yipan_handler.updateChairsideVideo)

module.exports = router

File: python_serve/routers/yipan.py
from fastapi import APIRouter, Request
from handlers.yipan_handler import (
    add_yipan, get_yipan_detail, update_yipan, list_yipan,
    start_chairside, complete_chairside, get_yipan_history, update_chairside_video
)

router = APIRouter(prefix="/api/yipan", tags=["yipan"])

@router.post("/add")
async def add_yipan_route(request: Request):
    data = await request.json()
    return add_yipan(data)

@router.post("/detail")
async def get_yipan_detail_route(request: Request):
    data = await request.json()
    return get_yipan_detail(data)

@router.post("/update")
async def update_yipan_route(request: Request):
    data = await request.json()
    return update_yipan(data)

@router.post("/list")
async def list_yipan_route(request: Request):
    data = await request.json()
    return list_yipan(data)

@router.post("/start")
async def start_chairside_route(request: Request):
    data = await request.json()
    return start_chairside(data)

@router.post("/complete")
async def complete_chairside_route(request: Request):
    data = await request.json()
    return complete_chairside(data)

@router.post("/history")
async def get_yipan_history_route(request: Request):
    data = await request.json()
    return get_yipan_history(data)

@router.post("/updateChairsideVideo")
async def update_chairside_video_route(request: Request):
    data = await request.json()
    return update_chairside_video(data)
